process/block/metablock.go

```diff
@@ -174,6 +174,8 @@ func NewMetaProcessor(arguments ArgMetaProcessor) (*metaProcessor, error) {
 
 	mp.shardsHeadersNonce = &sync.Map{}
 	mp.requestMissingHeadersFunc = mp.requestMissingHeaders
```

Comment:
after finishing this set of new functions with function pointers - we can refactor of giving it an internal component with an interface.

Reply:
Yes, it will be cumbersome to do this refactor, but could do it. Those new pointers that I've added will be removed anyway in future tasks, where the `shardCoordinator` will return correct value